From eae0536be815f732a1c11c892d5808ce4113f365 Mon Sep 17 00:00:00 2001
From: Alexey Edelev <alexey.edelev@qt.io>
Date: Wed, 29 Nov 2023 13:55:54 +0100
Subject: Use NO_GENERATE_CPP_EXPORTS explicitly

Use NO_GENERATE_CPP_EXPORTS explicitly for modules that don't need
the autogenerated exports header file.

Task-number: QTBUG-90492
Change-Id: Ifb068ca6eb404fefe4587346df8c511413f372f5
Reviewed-by: Qt CI Bot <qt_ci_bot@qt-project.org>
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
Reviewed-by: Moss Heim <moss.heim@qt.io>
---
 src/core/api/CMakeLists.txt          | 1 +
 src/pdfquick/CMakeLists.txt          | 1 +
 src/pdfwidgets/CMakeLists.txt        | 1 +
 src/webenginequick/CMakeLists.txt    | 1 +
 src/webenginequick/ui/CMakeLists.txt | 1 +
 src/webenginewidgets/CMakeLists.txt  | 1 +
 6 files changed, 6 insertions(+)

diff --git a/src/core/api/CMakeLists.txt b/src/core/api/CMakeLists.txt
index 87cb93eda..f2ceb2dfd 100644
--- a/src/core/api/CMakeLists.txt
+++ b/src/core/api/CMakeLists.txt
@@ -60,6 +60,7 @@ qt_internal_add_module(WebEngineCore
         Qt::Quick
     EXTRA_CMAKE_FILES
         "${CMAKE_CURRENT_LIST_DIR}/${INSTALL_CMAKE_NAMESPACE}WebEngineCoreDeploySupport.cmake"
+    NO_GENERATE_CPP_EXPORTS
 )
 
 if(MSVC)
diff --git a/src/pdfquick/CMakeLists.txt b/src/pdfquick/CMakeLists.txt
index 560817446..d57ce04aa 100644
--- a/src/pdfquick/CMakeLists.txt
+++ b/src/pdfquick/CMakeLists.txt
@@ -36,5 +36,6 @@ qt_internal_add_qml_module(PdfQuick
         Qt::Core
         Qt::Gui
         Qt::Qml
+    NO_GENERATE_CPP_EXPORTS
 )
 
diff --git a/src/pdfwidgets/CMakeLists.txt b/src/pdfwidgets/CMakeLists.txt
index 592409b66..41af017c0 100644
--- a/src/pdfwidgets/CMakeLists.txt
+++ b/src/pdfwidgets/CMakeLists.txt
@@ -15,4 +15,5 @@ qt_internal_add_module(PdfWidgets
         Qt::Gui
         Qt::Widgets
         Qt::Pdf
+    NO_GENERATE_CPP_EXPORTS
 )
diff --git a/src/webenginequick/CMakeLists.txt b/src/webenginequick/CMakeLists.txt
index 867a388eb..5f8344f08 100644
--- a/src/webenginequick/CMakeLists.txt
+++ b/src/webenginequick/CMakeLists.txt
@@ -55,6 +55,7 @@ qt_internal_add_qml_module(WebEngineQuick
         Qt::Qml
         Qt::Quick
         Qt::WebEngineCore
+    NO_GENERATE_CPP_EXPORTS
 )
 
 qt_internal_extend_target(WebEngineQuick CONDITION QT_FEATURE_webengine_webchannel
diff --git a/src/webenginequick/ui/CMakeLists.txt b/src/webenginequick/ui/CMakeLists.txt
index 87e3a0f90..ac960535e 100644
--- a/src/webenginequick/ui/CMakeLists.txt
+++ b/src/webenginequick/ui/CMakeLists.txt
@@ -37,6 +37,7 @@ qt_internal_add_qml_module(WebEngineQuickDelegatesQml
     NO_SYNC_QT
     PLUGIN_TARGET qtwebenginequickdelegatesplugin
     DEPENDENCIES QtQuickControls2
+    NO_GENERATE_CPP_EXPORTS
 )
 
 qt_internal_add_resource(qtwebenginequickdelegatesplugin "qtwebenginequickdelegatesplugin"
diff --git a/src/webenginewidgets/CMakeLists.txt b/src/webenginewidgets/CMakeLists.txt
index f9365adc9..dd11e48be 100644
--- a/src/webenginewidgets/CMakeLists.txt
+++ b/src/webenginewidgets/CMakeLists.txt
@@ -32,6 +32,7 @@ qt_internal_add_module(WebEngineWidgets
         Qt::Gui
         Qt::Widgets
         Qt::WebEngineCore
+    NO_GENERATE_CPP_EXPORTS
 )
 
 qt_internal_extend_target(WebEngineWidgets CONDITION QT_FEATURE_accessibility
-- 
cgit v1.2.3

